samples/test/downscoping-with-cab.test.js

```diff
@@ -0,0 +1,140 @@
+          availablePermissions: ['inRole:roles/storage.objectViewer'],
+          // Only objects starting with the specified prefix string in the object name
+          // will be allowed read access.
+          availabilityCondition: {
+            expression:
+              "resource.name.startsWith('projects/_/buckets/" +
+              `${bucketName}/objects/${objectName1}')`,
+          },
+        },
+      ],
+    },
+  };
+
+  it('should only have access to the object specified in the cab rule', async () => {
+    const projectId = await googleAuth.getProjectId();
+    const errorMessage =
+      'does not have storage.objects.get access to the Google Cloud Storage object.';
+
+    // Create the OAuth credentials (the consumer).
+    const oauth2Client = new OAuth2Client();
+    // We are defining a refresh handler instead of a one-time access
+    // token/expiry pair.
+    // This will allow the consumer to obtain new downscoped tokens on
+    // demand every time a token is expired, without any additional code
+    // changes.
+    oauth2Client.refreshHandler = async () => {
+      // Obtain an authenticated client via ADC.
+      const client = await googleAuth.getClient();
+
+      // Create a DownscopedClient with the credential access boundary defined
+      // above. This client will only be able to access the file contents of
+      // the specified object.
+      const cabClient = new DownscopedClient(client, cab);
+
+      // The common pattern of usage is to have a token broker pass the
+      // downscoped short-lived access tokens to a token consumer via some
+      // secure authenticated channel. For illustration purposes, we are
+      // generating the downscoped token locally.
+      const refreshedAccessToken = await cabClient.getAccessToken();
+      return {
+        access_token: refreshedAccessToken.token,
+        expiry_date: refreshedAccessToken.expirationTime,
+      };
+    };
+
+    const storageOptions = {
+      projectId,
+      authClient: {
+        sign: () => {
+          Promise.reject('unsupported');
+        },
+        getCredentials: async () => {
+          Promise.reject();
+        },
+        request: opts => {
+          return oauth2Client.request(opts);
+        },
+        authorizeRequest: async opts => {
+          opts = opts || {};
+          const url = opts.url || opts.uri;
+          const headers = await oauth2Client.getRequestHeaders(url);
+          opts.headers = Object.assign(opts.headers || {}, headers);
+          return opts;
+        },
+      },
+    };
+
+    const storage = new Storage(storageOptions);
+    // Test object1 can be downloaded.
+    const downloadFile = await storage
+      .bucket(bucketName)
+      .file(objectName1)
+      .download();
+    assert.strictEqual(downloadFile.toString('utf8').includes(CONTENT), true);
+
+    // Test object2 download fails due to no access.
+    try {
+      await storage.bucket(bucketName).file(objectName2).download();
```

Comment:
I would use `assert.rejects` here:

https://nodejs.org/api/assert.html#assertrejectsasyncfn-error-message

Reply:
I think about this way, would use assert.rejects if I can easily reproduce the entire error object and message since this assert.rejects API is comparing the generated error object with real error object. 

The error message formats like 
`byoid-test@cicpclientproj.iam.gserviceaccount.com does not have storage.objects.create access to the Google Cloud Storage object.`

The serviceAccount part needs to be fetched/read from given files, I don't feel like that's an stable/consistent way to set up the test, also following other language's style [python](https://github.com/googleapis/google-auth-library-python/blob/e6278a815895e050e57fc516f086b4bc89a0864a/system_tests/system_tests_sync/test_downscoping.py#L162)

Reply:
I think yoyu can just give `assert.rejects` a regex.

```
await assert.rejects(promise, /does not have storage\.objects\.create/);
```

The problem with the `try/catch` approach is that if `storage.bucket(bucketName).file(objectName2).download();` succeeds, you actually have broken behavior.

Reply:
I just realized `assert.rejects `doesn't work in chai which is the lib in package.json.
since you mentioned the broker behavior of download() what about this
```
    try {
      storage.bucket(bucketName).file(objectName2);
      throw new Error('should only have access to the object 2');
    } catch (e) {
      assert.strictEqual(e.message.includes(errorMessage), true);
    }
```
error is supposed to get caught in .file(objectName2) and compare to assert in errorMessage.

This way I don't need to break the style importing other packages just for this test.